lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
```

Comment:
I rescue it for ruby because RubyParser specifically raises it. If the other languages don't raise, then it shouldn't be pushed up.

I don't actually like `AnalyzerBase::RESCUABLE_ERRORS`... it's problematic in that you need to handle each and every type of exception that a given engine might raise. That's breaking responsibilities and should be pushed down. If necessary, a given engine should wrap an exception and raise an engine specific exception to be handled generically.